### PR TITLE
[shopsys] list of complaints in personal data

### PR DESCRIPTION
**Pull request by sspooky13**
#### Description, the reason for the PR
Print list of complaints in personal data page

#### Fixes issues
...

#### Check the appropriate checkboxes below, please

- [ ] [BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/) 
- [x] New feature 
- [x] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

## Summary by CodeRabbit

- **New Features**
	- Introduced a method for retrieving complaints by customer user, domain ID, and locale.
	- Added a "complaints" field to the Personal Data type in GraphQL, allowing users to view their complaints.
	- Enhanced the Personal Data Detail component to display user complaints in a structured format.

- **Bug Fixes**
	- Improved clarity and communication regarding complaints in the user interface through localization updates.

- **Documentation**
	- Updated upgrade notes to reflect the addition of complaints to the personal data page and query.

----
:globe_with_meridians: Live Preview:
  - https://pt-ssp-2650-personal-data-complaints.odin.shopsys.cloud
  - https://cz.pt-ssp-2650-personal-data-complaints.odin.shopsys.cloud

### CODE REVIEW

File: packages/framework/src/Model/Complaint/ComplaintFacade.php [hunk 1]
```diff
@@ -49,6 +50,20 @@ public function edit(int $id, ComplaintData $complaintData): void
         $this->em->flush();
     }
 
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser $customerUser
+     * @param int $domainId
+     * @param string $locale
+     * @return \Shopsys\FrameworkBundle\Model\Complaint\Complaint[]
+     */
+    public function getComplaintsByCustomerUserAndDomainIdAndLocale(
```
Comment (TomasLudvik) [inline packages/framework/src/Model/Complaint/ComplaintFacade.php:68]:
```suggestion
    public function getComplaintsByCustomerUserDomainIdAndLocale(
```

File: packages/framework/src/Model/Complaint/ComplaintRepository.php [hunk 2]
```diff
@@ -80,4 +81,27 @@ public function findById(int $id): ?Complaint
...
         return $this->getComplaintRepository()->find($id);
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser $customerUser
+     * @param int $domainId
+     * @param string $locale
+     * @return \Shopsys\FrameworkBundle\Model\Complaint\Complaint[]
+     */
+    public function getComplaintsByCustomerUserAndDomainIdAndLocale(
```
Comment (TomasLudvik) [inline packages/framework/src/Model/Complaint/ComplaintRepository.php:91]:
```suggestion
    public function getComplaintsByCustomerUserDomainIdAndLocale(
```

File: packages/framework/src/Model/Complaint/ComplaintRepository.php [hunk 3]
```diff
@@ -80,4 +81,27 @@ public function findById(int $id): ?Complaint
...
+        return $this->getComplaintRepository()->createQueryBuilder('cmp')
+            ->addSelect('c')->leftJoin('cmp.deliveryCountry', 'c')
+            ->addSelect('ct')->join('c.translations', 'ct', Join::WITH, 'ct.locale = :locale')
+            ->addSelect('cmpi')->join('cmp.items', 'cmpi')
+            ->where('cmp.customerUser = :customerUser')
+            ->andWhere('cmp.domainId = :domainId')
+            ->setParameter('customerUser', $customerUser)
+            ->setParameter('domainId', $domainId)
+            ->setParameter('locale', $locale)
+            ->getQuery()->getResult();
```
Comment (TomasLudvik) [inline packages/framework/src/Model/Complaint/ComplaintRepository.php]:
```suggestion
        return $this->getComplaintRepository()->createQueryBuilder('c')
            ->addSelect('cdc, ct, ci')
            ->leftJoin('c.deliveryCountry', 'cdc')
            ->join('c.translations', 'ct', Join::WITH, 'ct.locale = :locale')
            ->join('c.items', 'ci')
            ->where('c.customerUser = :customerUser')
            ->andWhere('c.domainId = :domainId')
            ->setParameter('customerUser', $customerUser)
            ->setParameter('domainId', $domainId)
            ->setParameter('locale', $locale)
            ->getQuery()->getResult();
```

File: project-base/app/schema.graphql [hunk 4]
```diff
@@ -935,11 +935,11 @@ type LoginResult {
 
 "One of the possible methods of the customer user login"
 enum LoginTypeEnum {
-  admin
+  web
   facebook
   google
   seznam
-  web
+  admin
```
Comment (TomasLudvik) [inline project-base/app/schema.graphql]: Do you have all composer dependencies updated? This seems weird

**Review by TomasLudvik — CHANGES_REQUESTED** [review]

File: packages/framework/src/Model/Complaint/ComplaintFacade.php (hunk 1 above)
Comment (sspooky13) [reply packages/framework/src/Model/Complaint/ComplaintFacade.php:68]: fixed https://github.com/shopsys/shopsys/pull/3433/commits/da1ccdd30769663d4b0ec08be9cb9e28cc23e29a

File: packages/framework/src/Model/Complaint/ComplaintRepository.php (hunk 2 above)
Comment (sspooky13) [reply packages/framework/src/Model/Complaint/ComplaintRepository.php:91]: fixed https://github.com/shopsys/shopsys/pull/3433/commits/da1ccdd30769663d4b0ec08be9cb9e28cc23e29a

File: packages/framework/src/Model/Complaint/ComplaintRepository.php (hunk 3 above)
Comment (sspooky13) [reply packages/framework/src/Model/Complaint/ComplaintRepository.php]: fixed https://github.com/shopsys/shopsys/pull/3433/commits/da1ccdd30769663d4b0ec08be9cb9e28cc23e29a

File: project-base/app/schema.graphql (hunk 4 above)
Comment (sspooky13) [reply project-base/app/schema.graphql]: yep, I ran `composer update` and then generated schema and nothing changed 🤷🤷‍♂️🤷‍♀️ Should I do something different?

Comment (TomasLudvik) [reply project-base/app/schema.graphql]: It is reverted when i run `make generate-schema` on my machine. Try to update all dependencies. Contact me if your problem persists.

**Review by TomasLudvik — APPROVED** [review]

**Review by TomasLudvik — APPROVED** [review]

**Review by pk16011990 — APPROVED** [review]

**Review by TomasLudvik — APPROVED** [review]

**Review by pk16011990 — APPROVED** [review]

**Pull request merged**